pkg/reporter/junit_reporter.go

```diff
@@ -0,0 +1,168 @@
+	Time       float32     `xml:"time,attr,omitempty"`
+	Timestamp  *time.Time  `xml:"timestamp,attr,omitempty"`
+	File       string      `xml:"file,attr,omitempty"`
+	Testcases  *[]Testcase `xml:"testcase,omitempty"`
+	Properties *[]Property `xml:"properties>property,omitempty"`
+	SystemOut  *SystemOut  `xml:"system-out,omitempty"`
+	SystemErr  *SystemErr  `xml:"system-err,omitempty"`
+}
+
+type Testcase struct {
+	XMLName         xml.Name         `xml:"testcase"`
+	Name            string           `xml:"name,attr"`
+	ClassName       string           `xml:"classname,attr"`
+	Assertions      int              `xml:"assertions,attr,omitempty"`
+	Time            float32          `xml:"time,attr,omitempty"`
+	File            string           `xml:"file,attr,omitempty"`
+	Line            int              `xml:"line,attr,omitempty"`
+	Skipped         *Skipped         `xml:"skipped,omitempty,omitempty"`
+	Properties      *[]Property      `xml:"properties>property,omitempty"`
+	TestcaseError   *TestcaseError   `xml:"error,omitempty"`
+	TestcaseFailure *TestcaseFailure `xml:"failure,omitempty"`
+}
+
+type Skipped struct {
+	XMLName xml.Name `xml:"skipped"`
+	Message string   `xml:"message,attr"`
+}
+
+type TestcaseError struct {
+	XMLName   xml.Name `xml:"error"`
+	Message   string   `xml:"message,omitempty"`
+	Type      string   `xml:"type,omitempty"`
+	TextValue string   `xml:",chardata"`
+}
+
+type TestcaseFailure struct {
+	XMLName   xml.Name `xml:"failure"`
+	Message   string   `xml:"message,omitempty"`
+	Type      string   `xml:"type,omitempty"`
+	TextValue string   `xml:",chardata"`
+}
+
+type SystemOut struct {
+	XMLName   xml.Name `xml:"system-out"`
+	TextValue string   `xml:",chardata"`
+}
+
+type SystemErr struct {
+	XMLName   xml.Name `xml:"system-err"`
+	TextValue string   `xml:",chardata"`
+}
+
+type Property struct {
+	XMLName   xml.Name `xml:"property"`
+	TextValue string   `xml:",chardata"`
+	Name      string   `xml:"name,attr"`
+	Value     string   `xml:"value,attr,omitempty"`
+}
+
+func (ts Testsuites) checkPropertyValidity() error {
+	for tsidx := range ts.Testsuites {
+		testsuite := ts.Testsuites[tsidx]
+		if testsuite.Properties != nil {
+			for pridx := range *testsuite.Properties {
+				property := (*testsuite.Properties)[pridx]
+				if property.Value != "" && property.TextValue != "" {
+					return fmt.Errorf("property %s in testsuite %s should contain value or a text value, not both",
+						property.Name, testsuite.Name)
+				}
+			}
+		}
+
+		if testsuite.Testcases != nil {
+			for tcidx := range *testsuite.Testcases {
+				testcase := (*testsuite.Testcases)[tcidx]
+				if testcase.Properties == nil {
+					continue
+				}
+				for propidx := range *testcase.Properties {
+					property := (*testcase.Properties)[propidx]
+					if property.Value != "" && property.TextValue != "" {
+						return fmt.Errorf("property %s in testcase %s should contain value or a text value, not both",
+							property.Name, testcase.Name)
```

Comment:
These lines are the exact same as 103-105 - consider breaking out into a separate function